### PR TITLE
Memory.load n

### PR DESCRIPTION
**Pull request by hurrikaanig**
refactoring Memory.load to Memory.load_n to load n bytes from memory

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

https://github.com/sayajin-labs/kakarot/issues/287

Issue Number: 287

## What is the new behavior?

-
-
-

### CODE REVIEW

File: src/kakarot/memory.cairo [hunk 1]
```diff
@@ -143,24 +143,26 @@ namespace Memory {
...
+        let offset_out_of_bounds = is_le(self.bytes_len, offset + n + 1);
         if (offset_out_of_bounds == 1) {
             let (local new_memory: felt*) = alloc();
             memcpy(dst=new_memory, src=self.bytes, len=self.bytes_len);
             Helpers.fill(
-                arr=new_memory + self.bytes_len, value=0, length=offset + 32 - self.bytes_len
+                arr=new_memory + self.bytes_len, value=0, length=offset + n - self.bytes_len
             );
-            let res: Uint256 = Helpers.bytes32_to_uint256(new_memory + offset);
+            let res: Uint256 = Helpers.bytes_i_to_uint256(new_memory + offset, n, zero);
```
Comment (Eikix) [inline src/kakarot/memory.cairo:161]: unrelated to your PR (in terms of scope) but why does `bytes_i_to_uint256` take in a third argument named res?

@LucasLvy any input:)?

File: src/kakarot/memory.cairo [hunk 2]
```diff
@@ -143,24 +143,26 @@ namespace Memory {
...
             return res;
         }
         with_attr error_message("Kakarot: Memory Error") {
-            let res: Uint256 = Helpers.bytes32_to_uint256(self.bytes + offset);
+            let res: Uint256 = Helpers.bytes_i_to_uint256(self.bytes + offset, n, zero);
```
[the remark this answers is not in the document]
Comment (hurrikaanig) [reply src/kakarot/memory.cairo]: even in tests ?

**Review by Eikix — CHANGES_REQUESTED** [review]
1. After seeing with @LucasLvy, we have confirmation that the `res` argument in `bytes_i_to_uint256` is useless. Could you add a commit that deletes this argument and refactors code that uses it?

2. There is a comment in the helper's code `// Check if i si inferiour to 32`, could you fix the unfortunate typo introduced there (not by you)?


Very cool PR overall, thanks a lot:)
Will be good to go right after

**Review by Eikix — APPROVED** [review]
Nice PR 🔥!

**Pull request merged**